manifests/server.pp

```diff
@@ -1,28 +1,42 @@
-# == Class: yum::server
+# @summary Manage a yum repository service
 #
-# Manage a yum repository service
+# @param contact_email
+#   Set email address for server administration contact.
+#   Will be used for ServerAdmin in Apache vhost configuration.
+#
+# @param docroot
+#   Set absolute path to document root. Will be used for DocumentRoot
+#   Apache vhost configuration.
+#
+# @param gpg_keys_path
+#   Set relative path to GPG keys directory which will be in $docroot
+#   directory. $docroot/$gpg_keys_path will be created and used.
+#
+# @param gpg_user_name
+#    Set user who signs the packages. Will be used as %_gpg_name in
+#    /root/.rpmmacros.
+#
+# @param yum_server
+#   Set servername for yum repository. Will be used for ServerName in
+#   Apache vhost configuration.
+#
+# @param yum_server_http_listen_ip
+#   Set listen IP for yum repository server. Will be used for VirtualHost
+#   in Apache vhost configuration.
 #
 class yum::server (
-  $contact_email             = 'root@localhost',
-  $docroot                   = '/opt/repos',
-  # gpg_keys_path is relative to $docroot
-  # ${docroot}/${gpg_keys_path}
-  $gpg_keys_path             = 'keys',
-  $gpg_user_name             = 'Root',
-  $yum_server                = 'yum',
-  $yum_server_http_listen_ip = $::ipaddress,
+  String $contact_email             = 'root@localhost',
+  Stdlib::Absolutepath $docroot     = '/opt/repos',
+  String $gpg_keys_path             = 'keys', # gpg_keys_path is relative to $docroot, ${docroot}/${gpg_keys_path}
+  String $gpg_user_name             = 'Root',
+  String $yum_server                = 'yum',
+  String $yum_server_http_listen_ip = $::ipaddress,
```

Comment:
Check out https://github.com/thrnio/puppet-ip for validating the IP address

Reply:
Really want to add another dependency just to check IP addesses ?

Reply:
yeah, we're already using it elsewhere and it seems to be the right way for network validation.